sky/packages/sky/lib/src/rendering/stack.dart

```diff
@@ -226,14 +230,112 @@ class RenderStack extends RenderBox with ContainerRenderObjectMixin<RenderBox, S
     defaultHitTestChildren(result, position: position);
   }
 
+  void paintStack(PaintingContext context, Offset offset);
+
   void paint(PaintingContext context, Offset offset) {
     if (_hasVisualOverflow) {
       context.canvas.save();
```

Comment:
@abarth / @chinmaygarde do we have a bug on file about the way paint functions like this one will break if any of the children use their own layer? I'm surprised we haven't run into this; surely it'd assert because the save stack isn't 0 when adding the children.